client/app/scripts/liveblog-edit/timeline.js

```diff
@@ -4,50 +4,14 @@ define([
 ], function(angular) {
     'use strict';
     TimelineController.$inject = ['api', '$scope', '$rootScope', 'notify', 'gettext',
-                                '$route', '$q', '$cacheFactory', 'userList', 'itemsService'];
+                                '$route', '$q', '$cacheFactory', 'userList', 'itemsService', 'postsService'];
     function TimelineController(api, $scope, $rootScope, notify, gettext,
-                                 $route, $q, $cacheFactory, userList, itemsService) {
-        var blog = {
-            _id: $route.current.params._id
-        };
-        $scope.posts = [];
-        $scope.totalPosts = 0;
-        $scope.timelineLoading = false;
-        $scope.isPostsEmpty = function() {
-            return $scope.posts.length === 0 && !$scope.timelineLoading;
-        };
-        $scope.postsCriteria = {
-            max_results: 10,
-            page: 1
-        };
-        function callbackCreator (i) {
-            return function(user) {
-                $scope.posts[i].original_creator_name = user.display_name;
-                if (user.picture_url) {
-                    $scope.posts[i].picture_url = user.picture_url;
-                }
-            };
-        }
-        $scope.getPosts = function() {
-            $scope.timelineLoading = true;
-            api('blogs/<regex(\"[a-f0-9]{24}\"):blog_id>/posts', blog).query($scope.postsCriteria).then(function(data) {
-                $scope.totalPosts = data._meta.total;
-                var posts = data._items;
-                // keep only opened posts
-                posts = posts.filter(function(post) {
-                    // check if undefined in order to support old model without this field.
-                    // (before the Draft-Posts Feature)
-                    return typeof(post.post_status) === 'undefined' || post.post_status === 'open';
-                });
-                for (var j = 0, post = {}; j < posts.length; j ++) {
-                    //build the new post system
-                    post = posts[j];
+                                 $route, $q, $cacheFactory, userList, itemsService, postsService) {
+
+        function retrievePosts() {
```

Comment:
can we make all the post processing inside the service?
here from my POV should only set the to scope.